hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py

```diff
@@ -40,32 +140,48 @@ def perform_enque_for_review(match_message: MatchMessage):
     logger.debug("enqued for review")
 
 
-possible_actions = {
-    "WRITEBACK_IN_REVIEW": perform_writeback_in_review,
-    "ENQUE_FOR_REVIEW": perform_enque_for_review,
-}
+class LabelWithConstraints(Label):
+    _KEY_CONSTRAINT = "KeyConstraint"
+
+    def __init__(self, value: str):
+        super(LabelWithConstraints, self).__init__(self._KEY_CONSTRAINT, value)
 
 
-class ActionLabel(Label):
-    def __init__(self, key: str, value: str):
-        if key != "Action":
-            raise Exception("ActionLabels must have a key Action")
+class ActionLabel(LabelWithConstraints):
+    _KEY_CONSTRAINT = "Action"
 
-        if value not in possible_actions.keys():
-            raise Exception(f"'%s' is not a valid Action" % value)
 
-        super(self.__class__, self).__init__(key, value)
+class ThreatExchangeReactionLabel(LabelWithConstraints):
+    _KEY_CONSTRAINT = "ThreatExchangeReaction"
+
+
+@dataclass
+class Action:
+    action_label: ActionLabel
+    priority: int
+    superseded_by: t.List[ActionLabel]
+
+
+@dataclass
+class ActionRule:
+    action_label: ActionLabel
+    must_have_labels: t.List[Label]
+    must_not_have_labels: t.List[Label]
 
 
 def perform_action(match_message: MatchMessage, action_label: ActionLabel):
-    for action, action_performer in possible_actions.items():
-        if action is action_label.value:
-            action_performer(match_message)
```

Comment:
Why was this removed? Did you mean to replace it with 
```
possible_actions[action_label.value](match_message)
```
from @schatten 's comment?

Reply:
@jeberl Nope removed on purpose. Actions are meant to be configured. At any time I should be able to go to the UI, add another rule with some new action label, along with corresponding configuration for action execution, and then there's a new rule that should be able to fire. So the concept of a hard-coded list / dict `possible_actions` is contrary to enabling that on-the-fly, configuration-based way of things working. But I might be missing something you were thinking, so let's talk.